storage/register_delta.go

```diff
@@ -68,6 +72,51 @@ func (r *RegisterDelta) GetUpdates() flow.RegisterEntries {
 	return entries
 }
 
-func (r *RegisterDelta) AllocateSlabIndex(_ []byte) (atree.SlabIndex, error) {
-	return atree.SlabIndex{}, nil
+func (r *RegisterDelta) AllocateSlabIndex(owner []byte) (atree.SlabIndex, error) {
```

Comment:
Can we extract it into a function that takes a ValueStore instance as argument? I think one day this function should just go to flow-go, and be reused everywhere.